internal/storage/image.go

```diff
@@ -420,6 +438,209 @@ func (svc *imageService) imageStatus(systemContext *types.SystemContext, unstabl
 	return &result, nil
 }
 
+func (svc *imageService) PullLocalManifestDigest(ctx context.Context, systemContext *types.SystemContext, userSpecifiedImage string) (string, error) {
+	imageParsed, err := reference.ParseNormalizedNamed(userSpecifiedImage)
```

Comment:
(As a matter of general policy, push string parsing as close to the top of the call stack as possible; use well-specified types like `RegistryImageReference` instead of strings. If you _must_ use strings, document the format as precisely as possible.)